src/ReferenceList.php

```diff
@@ -23,7 +27,7 @@
  * @author H. Snater < mediawiki@snater.com >
  * @author Thiemo Mättig
  */
-class ReferenceList extends HashableObjectStorage {
+class ReferenceList extends SplObjectStorage implements Comparable {
```

Comment:
I'm not sure if this is an improvement. I like it more if that bogus SPL stuff is in an other class.